### ISSUE TITLE
Videomass failed : HTTP Error 401: Unauthorized

### ISSUE THREAD

**Issue by tissatussa** [jeanslack/Videomass#13]
another issue i encountered twice today : while downloading all (!) videos of a large playlist (about 500 items), the process suddenly ended with many errors :

![videomass_unauthorized](https://user-images.githubusercontent.com/1109281/106211108-658c9e80-61c8-11eb-89d8-4d4de776b66c.png)

finally only 8 videos were downloaded, about 11 Gb (!) in total, and in both cases the process took several hours .. it started with downloading an "Access Token" :

![videomass_unauthorized2](https://user-images.githubusercontent.com/1109281/106211121-6f160680-61c8-11eb-8dbb-a41735024ac4.png)

i suspect the Access Token was expired after some time .. maybe this is a bug : you could try to download a new Access Token when this happens, but i have no clue why such error would occur ..

**Comment by jeanslack** [comment]
In relation to your comment on [#11](https://github.com/jeanslack/Videomass/issues/11#issuecomment-769459325) issue.

The most efficient way to check the real youtube-dl version in use by Videomass is through the View menu > youtube-dl > Version in use, because **1-** you may have multiple youtube-dl versions installed on your operating system, **2-** Videomass uses youtube-dl in different ways, two of them (mainly) are as local executable or as python library (the youtube_dl python package as in your case: /usr/local/lib/python3.8/dist-packages/youtube_dl). After doing this you can also check via the terminal as you did. 

I don't know how you installed youtube-dl (by package-manager of the your linux box? via pip?), **Because the errors above seem to come from an outdated version of youtube-dl.**
...Ok, you have updated youtube-dl (following instructions of some linux tech article) but are you sure that the whole package has also been updated? because I am not aware that it is possible to update the youtube-dl package to the latest version available through the package-manager of any linux distribution. Usually the youtube-dl versions in the linux repositories are obsolete.

I highly recommend you to remove that youtube-dl, and install the python3-pip package (on Debian/Ubuntu/Mint, `# apt install python3-pip`), then install the latest version of youtube-dl with pip: `$ pip3 install --user -U youtube-dl` (note the $ = as user not root)

Once this is done, the youtube-dl binary will be installed on "/home/roelof/.local/bin" while the package on "/home/roelof/.local/lib/python3.8/site-packages/youtube_dl" and you can update it without problems with the above command and without root privilegies.

**Comment by jeanslack** [comment]
As an alternative, if you don't want to do as mentioned above, open the Videomass Setup dialog, youtube-dl tab, and select: Use a local copy of youtube-dl updatable by Videomass 

Once restarted Videomass, when you click on the big button for downloading videos, Videomass will ask you for confirmation to install the latest youtube-dl release locally, which will be an executable of just over 2-3 Mb. At least so you can be sure you are using the latest version while leaving your system intact. 

Let me know .

**Comment by tissatussa** [comment]
today i succesfully re-installed youtube-dl, following your info .. i will be posting my procedure soon.  I also gathered some more tips while using Videomass, especially while downloading YT playlists.  I saw you created a new version .. nice !  I use it now.

**Comment by jeanslack** [comment]
..Ok
Remember to restart Videomass (if open) when installing/updating youtube-dl via pip command, as the youtube-dl package must be completely reloaded. Otherwise the old version will still be in use.

**Comment by tissatussa** [comment]
yes, i had multi youtube-dl files / versions on my Linux OS .. so i decided to examine this and clean my system and re-install the newest youtube-dl version .. here is what i did in detail :

the following files and dirs appeared ALL to be removed after uninstalling the youtube-dl version (2020.03.24-1 !?) which appeared in Synaptic. NOTE: after marking this youtube-dl for deletion, Synaptic also wanted to remove Videomass, so i let it do that ..

/var/lib/dpkg/info/youtube-dl.md5sums
/var/lib/dpkg/info/youtube-dl.list
/var/lib/dpkg/info/youtube-dl.postrm
/var/lib/dpkg/info/youtube-dl.preinst
/var/lib/dpkg/info/youtube-dl.postinst
/var/lib/dpkg/info/youtube-dl.prerm
/usr/bin/youtube-dl
/usr/share/doc/youtube-dl  # DIR without executable
/usr/share/man/man1/youtube-dl.1.gz
/usr/share/bash-completion/completions/youtube-dl  # script
/usr/share/fish/completions/youtube-dl.fish  # script
/usr/share/zsh/vendor-completions/_youtube-dl  # script

then, by searching file & dir names containing "youtube-dl" on my complete disk, the following files and dirs appeared NOT removed after uninstalling "youtube-dl" by Synaptic, so i deleted them all manually :

/home/roelof/.cache/youtube-dl
/root/.cache/youtube-dl
/etc/apt/sources.list.d/flexiondotorg-ubuntu-youtube-dl-gui-focal.list.save
/etc/apt/sources.list.d/flexiondotorg-ubuntu-youtube-dl-gui-focal.list
/etc/apt/trusted.gpg.d/flexiondotorg_ubuntu_youtube-dl-gui.gpg
/usr/local/etc/fish/completions/youtube-dl.fish
/usr/local/etc/bash_completion.d/youtube-dl.bash-completion
/usr/local/share/man/man1/youtube-dl.1  # text script "Automatically generated by Pandoc 2.1.3"
/usr/local/bin/youtube-dl
/usr/local/lib/python3.8/dist-packages/youtube_dl-2021.1.24.1.dist-info  # DIR contains no executables or scripts
/usr/local/lib/python3.8/dist-packages/youtube_dl  # DIR contains many Python scripts

then, by searching my complete disk, i discovered the following dir names containing "youtube_dl" (!? with underscore), so i deleted them all manually :

/usr/local/share/doc/youtube_dl
/usr/local/lib/python3.8/dist-packages/youtube_dl-2021.1.24.1.dist-info
/usr/local/lib/python3.8/dist-packages/youtube_dl

then, i installed youtube-dl again by your command (python3-pip was installed on my system) :
pip3 install --user -U youtube-dl

all went well .. version 2021.1.24.1 was installed.

after that, these files and dirs appeared to be installed :

/home/roelof/.local/etc/fish/completions/youtube-dl.fish
/home/roelof/.local/etc/bash_completion.d/youtube-dl.bash-completion
/home/roelof/.local/share/man/man1/youtube-dl.1
/home/roelof/.local/bin/youtube-dl (*)

also files in the following dirs (names containing 'youtube_dl') appeared to be installed :

/home/roelof/.local/share/doc/youtube_dl
/home/roelof/.local/lib/python3.8/site-packages/youtube_dl-2021.1.24.1.dist-info
/home/roelof/.local/lib/python3.8/site-packages/youtube_dl

*) the Python script /home/roelof/.local/bin/youtube-dl contains the code line "from youtube_dl import main" which seems to refer to the newly installed dir /home/roelof/.local/lib/python3.8/site-packages/youtube_dl

**Comment by jeanslack** [comment]
The new installation of youtube-dl via pip3 looks good. Use the same command to update new versions of youtube-dl.

> ...after marking this youtube-dl for deletion, Synaptic also wanted to remove Videomass

That's why I decided to remove youtube-dl from required dependencies in the next versions of Videomass (only for some packages of course). The user will be free to install youtube-dl as he prefers. I think it may be the best choice.
